engine/minskyTensorOps.cc

```diff
@@ -447,27 +446,124 @@ namespace minsky
     }
   };
 
-  class RavelTensor: public CachedTensorOp
+  class RavelTensor: public civita::DimensionedArgCachedOp
```

Comment:
RavelTensor is not a DimensionedArgCachedOp. A DimensionedArgCachedOp is one that applies to a specific dimension, as specified by the operation "axis" attribute. Eg reduction along a specific dimension (or axis).

Reply:
Fixed this. Thanks for the explanation.